ethcore/src/miner/miner.rs

```diff
@@ -1451,8 +1456,11 @@ impl miner::MinerService for Miner {
 						service_transaction_checker.as_ref(),
 					);
 					queue.cull(client);
-					if is_internal_import {
-						chain.update_sealing();
+					if cloned_tx_queue.all_transaction_hashes().len() > 0 {
+						if !chain.update_sealing() {
```

Comment:
But this is not really a desired behavior for all engines. It's just a property of `InstantSeal` that it should produce blocks as long as there are transactions in the pool.

Adding this snip of code here will actually cause all engines to produce pending blocks way too often and that's really wasteful. I think we should rather make this behavior engine-specific.